google/api_core/path_template.py

```diff
@@ -193,3 +193,58 @@ def validate(tmpl, path):
     """
     pattern = _generate_pattern_for_template(tmpl) + "$"
     return True if re.match(pattern, path) is not None else False
+
+def transcode(http_options, **request_kwargs):
+    """Transcodes a grpc request pattern into a proper HTTP request following the rules outlined here,
+       https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L44-L312
+
+        Args:
+            http_options (list(dict)): A list of dicts which consist of these keys,
+                'method'    (str): The http method
+                'uri'       (str): The path template
+                'body'      (str): The body field name (optional)
+                (This is a simplified representation of the proto option `google.api.http`)
+            request_kwargs (dict) : A dict representing the request object
+
+        Returns:
+            dict: The transcoded request with these keys,
+                'method'        (str)   : The http method
+                'uri'           (str)   : The expanded uri
+                'body'          (dict)  : A dict representing the body (optional)
+                'query_params'  (dict)  : A dict mapping query parameter variables and values
+
+        Raises:
+            ValueError: If the request does not match the given template.
+    """
+    request = {}
+    for http_option in http_options:
+
+        # Assign path
+        uri_template = http_option['uri']
+        path_fields = [match.group('name') for match in _VARIABLE_RE.finditer(uri_template)]
+        path_args = {field: request_kwargs.get(field) for field in path_fields}
```

Comment:
I think this will be a problem with path-type field names, like 'resource.type'.  If the request has a field 'resource' which is a message, containing a field 'type', the spec allows '{resource.type=*}'.  In this case, you need something like request_kwargs.get('request').get('type'), except more dynamic, something like:
val = request_kwargs
for name_seg = field.split('.'):
  val = val.get(name_seg)
but also look out for Nones, of course,  Maybe put it in a function.
There should also be test cases for this.

Reply:
To clarify:  consider this python snippet:
```
>>> x = {'a':{'b':'abc'},'c':'def'}
>>> def show_args(**kwargs):
...   print(kwargs)
... 
>>> show_args(**x)
{'a': {'b': 'abc'}, 'c': 'def'}
```
Note that you don't get anything like `a.b`, you get a nested dict (which makes sense).  So you have to navigate this in code.

Reply:
All set - I added the functions `get_field` and `delete_field` to address this problem. I also added unit tests.